R/shared-methods.R

```diff
@@ -55,114 +55,74 @@ check_country_available <- function(country = character(), level = 1,
-  self$set_region_codes()
-
-  if (!missing(regions)) {
-    self$target_regions <- regions
-  }
-
-  if (!is.null(self$level_data_urls[[self$level]])) {
-    if (length(self$common_data_urls) > 0) {
-      self$data_urls <- unlist(merge(
-        self$level_data_urls[[self$level]],
-        self$common_data_urls
-      ))
-    } else {
-      self$data_urls <- self$level_data_urls[[self$level]]
-    }
-  } else {
-    self$data_urls <- self$common_data_urls
-  }
-
-  if (get) {
-    self$get()
-  }
-}
-
 #' R6 Class containing non-country specific methods
 #'
 #' @description Acts as parent class for individual country objects,
 #' allowing them to access general methods.
 #'
 #' @details All countries have shared methods for extracting region codes,
-#' downloading, processing, and returning data.
+#' downloading, processing, and returning data. These functions are contained
+#' within this parent class and so are accessible by all countries which
+#' inherit from here. Individual countries can overwrite any functions or
+#' fields providing they define a method with the same name.
 DataClass <- R6::R6Class(
   "DataClass",
   public = list(
     #' @field country name of country to fetch data for
     country = "",
-    #' @field data data frame for requested region
+    #' @field data Once initialised, a list of named dataframes: raw
+    #' (list of named raw dataframes) clean (cleaned data) and processed
+    #' (processed data). Data is accessed using `$data`.
     data = NULL,
     #' @field supported_levels A list of supported levels.
     supported_levels = list("1"),
     #' @field supported_region_names A list of region names in order of level.
     supported_region_names = list("1" = NA),
     #' @field supported_region_codes A list of region codes in order of level.
     supported_region_codes = list("1" = NA),
-    #' @field region_name string Name for the codes column, e.g. 'iso_3166_2'
+    #' @field region_name string Name for the region column, e.g. 'region'.
+    #' This field is filled at initialisation with the region name for the
+    #' specified level (supported_region_names$level).
     region_name = NULL,
     #' @field code_name string Name for the codes column, e.g. 'iso_3166_2'
+    #' Filled at initialisation with the code name associated with the
+    #' requested level (supported_region_codes$level).
     code_name = NULL,
     #' @field codes_lookup string or tibble Region codes for the target country
+    #' filled by country specific codes in
+    #' \href{#method-set_region_codes}{\code{set_region_codes()}}
     codes_lookup = list(),
-    #' @field data_urls List of named links to raw data.
+    #' @field data_urls List of named common and shared url links to raw data.
+    #' Prefers common if there is a name conflict.
     data_urls = list(),
     #' @field common_data_urls List of named links to raw data that are common
     #' across levels. The first entry should be named main.
```

Comment:
is there a reason we are enforcing this now apart from consistency? I don't mind so just a comment

Reply:
Just for consistency but doesn't have to be